src/app/component/ProductCard/ProductCard.container.js

```diff
@@ -160,36 +162,28 @@ export class ProductCardContainer extends PureComponent {
     }
 
     _getAvailableVisualOptions() {
-        const { product: { configurable_options = [] } } = this.props;
-
-        return Object.values(configurable_options).reduce((acc, { attribute_options = {}, attribute_values }) => {
-            const visualOptions = Object.values(attribute_options).reduce(
-                (acc, option) => {
-                    const {
-                        swatch_data,
-                        label,
-                        value: attrValue
-                    } = option;
-
-                    const { type, value } = swatch_data || {};
-
-                    if (
-                        type === '1'
-                        && attribute_values.includes(attrValue)
-                    ) {
-                        acc.push({ value, label });
-                    }
-
-                    return acc;
-                }, []
-            );
-
-            if (visualOptions.length > 0) {
-                return [...acc, ...visualOptions];
-            }
-
-            return acc;
-        }, []);
+        const { product: { configurable_options = {} } } = this.props;
+
+        if (Object.keys(configurable_options).length === 0) {
+            return [];
+        }
+
+        return Object.values(Object.values(configurable_options)[0].attribute_options).reduce(
```

Comment:
Please split this `Object.values(Object.values(configurable_options)[0].attribute_options)` declaration into multiple